### ISSUE TITLE
xrandr configuration relies on exactly one specific monitor setup

### ISSUE THREAD

**Issue by HunterThueson** [HunterThueson/dots#2]
xorg.nix currently declares my correct monitor setup, but if I were to do any of the following:

a) unplug one or both of my monitors
b) plug a monitor into a different port on the GPU
c) change the relative location(s) of one or more monitors
etc. etc.

my entire system configuration would either fail to start up entirely or would fall back to `xrandr`'s autogenerated (and probably incorrect) configuration(s). I still need to test what exactly would happen, but either option would be bad.

Solution: write a more comprehensive .nix file that analyzes the monitor configuration on startup and automatically generates an `xrandr` /  `X11` configuration to match. It should be configurable via a short line (or at least a few short lines) in `configuration.nix`. For example, adding these lines:

```
monitor_config = {
    4K-144hz = "right"
    2K-144hz = "left_centered"    # align with center of 4K monitor
};
```

should result in a working configuration regardless of which GPU port each monitor is plugged into. The exact syntax is TBD and names are subject to change, this is just the general idea.

**Comment by HunterThueson** [comment]
2022-10-22: I decided to test what would happen if I swapped which GPU ports the monitors were plugged into. As expected, the `xrandr` configuration
_did_ go all wonky. More specifically, the 4k monitor looked fantastic, but the 2k monitor on the left was displaying two different SDDM login screens
split vertically (horizontally? One on top of the other is what I mean). The top login screen seemed to be the one that was 'focused' when I started
entering my password, but it was covered up (starting about halfway down) by the bottom login screen. This is essentially the same bug that prompted me
to write the `xorg.nix` script in the first place. The only difference is that it is now happening on the 2k monitor instead of the 4k monitor.

Interestingly, swapping GPU ports fixes #16 -- it seems like the BIOS info and boot log will automatically display on the "top"/"first" port of the GPU.
It's also worth noting that, at least for the 1-2 minutes I spent using this alternate setup, the UI scaling issues from #10 seemed to have been fixed as
well (although I would need to do more thorough testing in order to know for sure).

I think I'll do a temporary rewrite of `xorg.nix` to set up `xrandr` properly for this new configuration. It won't fix this issue, since the script won't
adapt to different monitor configurations, but fixing #10 and #16 seems worth the small amount of extra effort it will take to write a temporary patch.

**Comment by HunterThueson** [comment]
After spending the last several weeks learning `xrandr` and Nvidia driver settings and rewriting `xorg.nix` for a new setup, I've come to the conclusion that trying to write `xorg.nix` in a way that works for any monitor setup is a long way off at best and futile at worst. It's better just to write a new configuration for each new monitor setup. That being said, I'm adding lots of notes about each part of the process so that the next time I switch to a new setup I won't have to re-learn everything. That's about the best I can do at this point.
